### PR TITLE
Added verbose errors for "parsing error", uncompressed reader for algo=0

### PR DESCRIPTION
**Pull request by skunkworker**
When a `parsing error` occurs there is no verbosity on what the actual error is, this makes the error message more verbose.

And when a packet has `algo=0` use a regular io.reader interface without decompression.

### CODE REVIEW

**Comment by film42** [comment]
We found this because a client of ours was sending us uncompressed encrypted files. Our clients usually have a lot of legacy systems preventing them from doing something like compressing files. Would love for this use-case to be merged. I think RFC 4880 says this MUST be implemented, but I'm not sure if ProtonMail has other reasons for not implementing this:

```
9.3.  Compression Algorithms

       ID           Algorithm
       --           ---------
       0          - Uncompressed
       1          - ZIP [RFC1951]
       2          - ZLIB [RFC1950]
       3          - BZip2 [BZ2]
       100 to 110 - Private/Experimental algorithm

   Implementations MUST implement uncompressed data.  Implementations
   SHOULD implement ZIP.  Implementations MAY implement any other
   algorithm.
```

File: openpgp/read.go [hunk 1]
```diff
@@ -223,9 +224,9 @@ FindKey:
 	if err := packets.Push(decrypted); err != nil {
 		return nil, err
 	}
-	mdFinal, sensitiveParsingErr := readSignedMessage(packets, md, keyring, config)
```
Comment (skunkworker) [inline openpgp/read.go:226]: I renamed this error to be more consistent with usages later in the file.

**Comment by twiss** [comment]
Hi :wave: Thanks for the PR.

Re. the uncompressed algorithm, it's interesting that you ran into that. Typically, when creating uncompressed encrypted messages, you wouldn't create a compressed data packet at all, and just put the literal data packet there directly. My interpretation of RFC4880 is that *that* must be supported, not that a compressed data packet with uncompressed data packet must be supported (however, I'm not sure what the original intention was). That interpretation is also codified in the [crypto-refresh](https://datatracker.ietf.org/doc/html/draft-ietf-openpgp-crypto-refresh-06#section-13.3.1.1). I also never encountered such packets before. But, if they're out there, we should probably add support for it, yeah.

Instead, regarding the error messages, the reason why they're called `sensitiveParsingErr` is because we shouldn't expose the error message in certain cases, to avoid decryption oracles. But, we could think about adding a "debug mode" someday that exposes them. But I would separate that from the rest of this PR.

**Comment by skunkworker** [comment]
@twiss I created a separate MR for just the `algo=0` with an example. https://github.com/ProtonMail/go-crypto/pull/121

**Comment by skunkworker** [comment]
As for the `sensitiveParsingErr` being more verbose, it would be advantageous to have because we commonly get strange and archaic pgp payloads from clients who are using all manner of GPG libraries. In order to track down this specific issue with the `algo=0` I had to recompile a checking tool that printed more verbose errors, as all we would see was `
parsing error"`.

**Pull request closed without merging**